File: C#/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int>ValToIndex = new ();
        int arrayLength = nums.Length;
        for(int i=0;i<arrayLength;i++)
        {
            if (ValToIndex.ContainsKey(target-nums[i]))
            {
                return new[] {ValToIndex[target-nums[i]],i};
            }
            ValToIndex[nums[i]] = i;
        }
        return Array.Empty<int>();
    }
}

File: JavaScript/Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashMap = new Map();
    let result = [];
    var arrayLength = nums.length;
    for(let i=0;i<arrayLength;i++)
    {
        if(hashMap.has((target - nums[i])))
        {
            result.push(hashMap.get((target-nums[i])));
            result.push(i);
            return result;
        }
        hashMap.set(nums[i],i);
    }
    return result;
};
